### PR TITLE
Make `skip_on_exception` a signature-preserving decorator; issue 363

### LINKED ISSUE #363: Make "skip_on_exception" a signature-preserving decorator.
Currently `skip_on_exception`  is not a signature-preserving decorator, at least in Python 2.7. When writing tests one needs to use a workaround to  parametrize `skip_on_exception` decorator see https://github.com/pydata/pandas-datareader/pull/357 for discussion.

### PR DESCRIPTION
**Pull request by OlegShteynbuk**
closes #363  

using wrapt package

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -7,4 +7,4 @@ commands=
 deps=
     pytest
     pytest-cov
-	wrapt
\ No newline at end of file
+	wrapt
```
Comment (bashtage) [inline tox.ini]: Spacing error here

Comment (OlegShteynbuk) [inline tox.ini]: changed, but the tests passed even without new line

File: tox.ini [hunk 2]
```diff
@@ -7,3 +7,4 @@ commands=
 deps=
     pytest
     pytest-cov
+	wrapt
```
Comment (bashtage) [inline tox.ini]: Should be

```

deps=
  pytest
  pytest-cov
  wrapt
```
so that the left column is vertically aligned.

Comment (OlegShteynbuk) [inline tox.ini]: thanks, it was a tab, replaced with spaces, probably need to set it up in my editor

File: pandas_datareader/_testing.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 Utilities for testing purposes.
```
Comment (jreback) [inline pandas_datareader/_testing.py]: don't change the mode of the file, use

``git update-index --chmod -x <file>`` to put it back

**Comment by jreback** [comment]
minor change, otherwise lgtm. ping on green.

**Comment by jreback** [comment]
thanks!

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas-datareader/commit/6cce5f18d52be802c7245c8a28d534236a9e2b24

File: pandas_datareader/_testing.py (hunk 3 above)
Comment (OlegShteynbuk) [inline pandas_datareader/_testing.py]: was offline for a while, it's already merged.

**Pull request closed without merging**